zebra-consensus/src/verify/block.rs

```diff
@@ -50,6 +50,24 @@ pub(crate) fn node_time_check(
     }
 }
 
+/// [3.10]: https://zips.z.cash/protocol/protocol.pdf#coinbasetransactions
+pub(crate) fn coinbase_check(block: &Block) -> Result<(), Error> {
+    if block.coinbase_height().is_some() {
+        // No additional coinbase transactions allowed
+        if block.transactions.len() > 1 {
+            let txs = block.transactions.iter().skip(1);
+            for t in txs {
+                if t.is_coinbase().is_some() {
+                    return Err("coinbase tx found not in the first position".into());
```

Comment:
Suggestion: We try to use `?` to get good backtraces:
```suggestion
                    Err("coinbase tx found not in the first position")?
```

Reply:
I changed to this but also noticed clippy suggests `return Err("coinbase input found in additional transaction".into())`

Reply:
I disabled that clippy suggestion in zebra-consensus in #524 in 64c53b74102c282e75965f7bb7eb1025afa49ae8. I should be able to merge that fix soon.

We think that suggestion is a mistake, because it makes it harder to diagnose failures. Maybe we'll get clippy fixed some time :-)

Reply:
Got it, thanks.